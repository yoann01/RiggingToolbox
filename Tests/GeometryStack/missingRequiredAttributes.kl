
require RiggingToolbox;

operator entry(){

  GeometryStack stack();
  stack.addGeometryOperator(PolygonMeshSphereGenerator(2.0, 1, true, false));
  stack.addGeometryOperator(PushModifier());

  EvalContext context();
  // Causes and erorr because the PushModifier requires normals, but the sphere doesn't have them. 
  stack.evaluate(context);
}

