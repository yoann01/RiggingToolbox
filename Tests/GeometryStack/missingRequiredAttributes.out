Error: Cannot evaluate 'PushModifier'. Geometry missing required attributes:["normals"]
KL stack trace:
[ST] 1 kl.internal.String.SetErrorDataPtrAndLength.AS0()
[ST] 2 function.setError.R.ST()
[ST] 3 method.evaluate.L.UO_GeometryStack.R.OO_EvalContext() GeometryStack.kl:145
[ST] 4 operator.entry() missingRequiredAttributes.kl:12
[ST] 5 kl.internal.entry.stub.cpu()
